### ISSUE TITLE
LEFT-TO-RIGHT MARK sometimes causing a crash

### ISSUE THREAD

**Issue by JulienPalard** [Alir3z4/html2text#208]
Hi, using `html2text 2018.1.9` and `Python 3.7.0`, I'm having an issue with the `U+200E LEFT-TO-RIGHT MARK` caracter:

```
$ html2text <<< '<html> <body> <b>b</b>&#8206; </body> </html>'
Traceback (most recent call last):
  File "/home/mdk/.venvs/googlesearchd/bin/html2text", line 11, in <module>
    sys.exit(main())
  File "/home/mdk/.venvs/googlesearchd/lib/python3.7/site-packages/html2text/cli.py", line 324, in main
    wrapwrite(h.handle(data))
  File "/home/mdk/.venvs/googlesearchd/lib/python3.7/site-packages/html2text/__init__.py", line 149, in handle
    self.feed(data)
  File "/home/mdk/.venvs/googlesearchd/lib/python3.7/site-packages/html2text/__init__.py", line 146, in feed
    HTMLParser.HTMLParser.feed(self, data)
  File "/home/mdk/.pyenv/versions/3.7.0-debug/lib/python3.7/html/parser.py", line 111, in feed
    self.goahead(0)
  File "/home/mdk/.pyenv/versions/3.7.0-debug/lib/python3.7/html/parser.py", line 204, in goahead
    self.handle_charref(name)
  File "/home/mdk/.venvs/googlesearchd/lib/python3.7/site-packages/html2text/__init__.py", line 186, in handle_charref
    self.handle_data(self.charref(c), True)
  File "/home/mdk/.venvs/googlesearchd/lib/python3.7/site-packages/html2text/__init__.py", line 802, in handle_data
    and re.match(r'[^\s.!?]', data[0])
IndexError: string index out of range
```

**Comment by JulienPalard** [comment]
I think the big problem is `charref` can easily return the empty string, (`return ''` obviously, and `return unifiable_n[c]` both can return the empty string too).

But `handle_data` receiving the result of `charref` hits `data[0]` as soon as its line 6.

**Comment by Alir3z4** [comment]
Did https://github.com/Alir3z4/html2text/pull/206 fixed this ?

**Comment by JulienPalard** [comment]
No:

```
$ git rev-parse --short HEAD
a9a6133
$ pip install -e .
Obtaining file:///home/mdk/clones/html2text
Installing collected packages: html2text
  Running setup.py develop for html2text
Successfully installed html2text
$ which html2text 
/home/mdk/.venvs/3.7.1/bin/html2text
$ html2text <<< '<html> <body> <b>b</b>&#8206; </body> </html>'
Traceback (most recent call last):
  File "/home/mdk/.venvs/3.7.1/bin/html2text", line 11, in <module>
    load_entry_point('html2text', 'console_scripts', 'html2text')()
  File "/home/mdk/clones/html2text/html2text/cli.py", line 337, in main
    wrapwrite(h.handle(data))
  File "/home/mdk/clones/html2text/html2text/__init__.py", line 151, in handle
    self.feed(data)
  File "/home/mdk/clones/html2text/html2text/__init__.py", line 148, in feed
    HTMLParser.HTMLParser.feed(self, data)
  File "/home/mdk/.pyenv/versions/3.7.1/lib/python3.7/html/parser.py", line 111, in feed
    self.goahead(0)
  File "/home/mdk/.pyenv/versions/3.7.1/lib/python3.7/html/parser.py", line 204, in goahead
    self.handle_charref(name)
  File "/home/mdk/clones/html2text/html2text/__init__.py", line 188, in handle_charref
    self.handle_data(self.charref(c), True)
  File "/home/mdk/clones/html2text/html2text/__init__.py", line 813, in handle_data
    and re.match(r'[^\s.!?]', data[0])
IndexError: string index out of range
```

**Comment by Alir3z4** [comment]
Would you be able to submit patch for this problem ?

**Comment by jdufresne** [comment]
@JulienPalard I've proposed a fix for this in #255. If you have the time, could you test it against your real life use case? Thanks.
